src/main/java/eu/eutampieri/catacombs/model/Entity.java

```diff
@@ -104,35 +104,36 @@ public final int getSize() {
      */
     protected void move() {
 
-        final int maxMovementUp = speedY;
-        final int maxMovementRight = speedX;
-        final int maxMovementDown = speedY;
-        final int maxMovementLeft = speedX;
-
+        int dx = 0, dy = 0;
         if (up) {
-            if (!isUpCollision(maxMovementUp)) {
-                hitBox.move(0, -maxMovementUp);
-            }
+            dy = -speedY;
             face = Direction.UP;
+            while (isUpCollision(Math.abs(dy))) {
+                dy++;
+            }
```

Comment:
For @MyK00L : @jackmounts fears that these could significantly spin up the CPU

Reply:
Loop goes on to a max of the entity movement speed